tests/plugins/lti_consumer/test_cms_plugins.py

```diff
@@ -0,0 +1,172 @@
+"""Testing DjangoCMS plugin declaration for Richie's lti consumer plugin."""
+from unittest import mock
+
+from django.test import TestCase, override_settings
+from django.test.client import RequestFactory
+
+from cms.api import add_plugin
+from cms.models import Placeholder
+from cms.plugin_rendering import ContentRenderer
+from django.conf import settings
+
+from richie.plugins.lti_consumer.cms_plugins import LTIConsumerPlugin
+from richie.plugins.lti_consumer.factories import LTIConsumerFactory
+
+
+@override_settings(
+    LTI_CONSUMER_SECRETS={
+        "marsha": {
+            "oauth_consumer_key": "InsecureOauthConsumerKey",
+            "shared_secret": "InsecureSharedSecret",
+        }
+    },
+    LTI_CONSUMER={
+        "display_name": "Marsha Video",
+        "pattern": "https://marsha\.education.*",
+        "is_launch_url_regex": True,
+        "automatic_resizing": True,
+        "inline_ratio": 0.5625,
+    },
+)
+class LTIConsumerPluginTestCase(TestCase):
+    """Test suite for the lti consumer plugin."""
+
+    def setUp(self):
+        self.maxDiff = None
+
+    def test_cms_plugins_lti_consumer_context_and_html(self):
+        """
+        Instanciating this plugin with an instance should populate the context
+        and render in the template.
+        """
+        placeholder = Placeholder.objects.create(slot="test")
+
+        lti_consumer = LTIConsumerFactory()
+
+        model_instance = add_plugin(
+            placeholder,
+            LTIConsumerPlugin,
+            "en",
+            url=lti_consumer.url,
+            course=lti_consumer.course,
+        )
+        plugin_instance = model_instance.get_plugin_class_instance()
+        context = plugin_instance.render({}, model_instance, None)
+
+        # Check if "instance" is in context
+        self.assertIn("instance", context)
+
+        # Check if parameters, generated by the factory, are correctly set in "instance" of context
+        self.assertEqual(context["instance"].url, lti_consumer.url)
+
+        # Get generated html for lti consumer url
+        renderer = ContentRenderer(request=RequestFactory())
+        html = renderer.render_plugin(model_instance, {})
+
+        # Check rendered url is correct after save and sanitize
+        self.assertIn(lti_consumer.url, html)
+
+    def test_cms_plugins_lti_auth_parameters(self):
+        lti_consumer = LTIConsumerFactory()
+        expected_auth_parameters = {
+            "lti_message_type": settings.LTI_CONSUMER.get("display_name"),
+            "lti_version": "LTI-1p0",
+            "resource_link_id": str(lti_consumer.id),
+            "context_id": lti_consumer.course.code,
+            "user_id": "richie",
+            "lis_person_contact_email_primary": "fun.dev@fun-mooc.fr",
+            "roles": "instructor",
```

Comment:
It should only be instructor when in edit mode

Reply:
It's fixed now